src/LinAlg/hiopVectorCuda.cpp

```diff
@@ -287,11 +278,8 @@ void hiopVectorCuda::startingAtCopyFromStartingAt(int start_idx_dest,
 /// @brief Copy `this` to double array, which is assumed to be at least of `n_local_` size.
 void hiopVectorCuda::copyTo(double* dest) const
 {
-  cudaError_t cuerr = cudaMemcpy(dest,
-                                 data_,
-                                 (n_local_)*sizeof(double),
-                                 cudaMemcpyDeviceToDevice);
-  assert(cuerr == cudaSuccess);
+  auto b = exec_space_.copy(dest, data_, n_local_);
+  assert(b);
 }
```

Comment:
This const method requires me to change the signature of function `do_it` [(see here)](https://github.com/LLNL/hiop/blob/45909b3c1dc4ae669b4833327e10d0e1f161315e/src/ExecBackends/ExecSpace.hpp#L477). I think this change is safe for now, since `hwb_dest` has never been changed in the code.

Reply:
I understand. Since that `copyTo` method works will raw pointers, it will be removed soon. Please remove the `const` in `TransferImpl` and use a workaround in this method, e.g. const cast as in (https://github.com/LLNL/hiop/blob/d047558d69065c2d220d76d169188f7231826b58/src/LinAlg/hiopVectorRajaImpl.hpp#L844)